torch/csrc/autograd/record_function.cpp

```diff
@@ -218,7 +218,7 @@ void _runBeforeCallbacks(RecordFunction* rf, const std::string& funcName) {
 }
 
 RecordFunction::RecordFunction(RecordScope scope) : scope_(scope) {
-  if (manager().hasCallbacks()) {
+  if (manager().hasCallbacks() && at::_tls_is_record_function_enabled()) {
```

Comment:
By default, if there is no guard, this will always be true right?

Reply:
correct